File: ControleFutebolWeb/Models/Api/JogoDtos.cs
namespace ControleFutebolWeb.Models.Api
{
    public class JogoResumoDto
    {
        public int Id { get; set; }
        public DateTime? Data { get; set; }
        public int CompeticaoId { get; set; }
        public string? CompeticaoNome { get; set; }
        public int TimeCasaId { get; set; }
        public string TimeCasaNome { get; set; } = string.Empty;
        public string? TimeCasaEscudoUrl { get; set; }
        public int TimeVisitanteId { get; set; }
        public string TimeVisitanteNome { get; set; } = string.Empty;
        public string? TimeVisitanteEscudoUrl { get; set; }
        public int? PlacarCasa { get; set; }
        public int? PlacarVisitante { get; set; }
        public string? Status { get; set; }
        // true quando o usuário logado marcou este jogo como analisado (mesma
        // regra do painel /Jogos/Hoje da web).
        public bool AnalisadoPorMim { get; set; }
    }

    public class JogoDetalheDto : JogoResumoDto
    {
        public string? Estadio { get; set; }
        public string? Arbitro { get; set; }
        public int? PenaltisCasa { get; set; }
        public int? PenaltisVisitante { get; set; }

        public List<EscalacaoJogadorDto> EscalacaoCasa { get; set; } = new();
        public List<EscalacaoJogadorDto> EscalacaoVisitante { get; set; } = new();
        public List<GolJogoDto> Gols { get; set; } = new();
        public List<CartaoJogoDto> Cartoes { get; set; } = new();
        // Estatísticas de partida por time (posse, finalizações etc.), extraídas
        // do EstatisticasJson importado da api-football.
        public List<EstatisticaTimeJogoDto> EstatisticasTimes { get; set; } = new();
    }

    public class EscalacaoJogadorDto
    {
        public int JogadorId { get; set; }
        public string Nome { get; set; } = string.Empty;
        public string? Posicao { get; set; }
        public bool Titular { get; set; }
        public string? FotoUrl { get; set; }
    }

    public class GolJogoDto
    {
        public int JogadorId { get; set; }
        public string JogadorNome { get; set; } = string.Empty;
        public int Minuto { get; set; }
        public bool Contra { get; set; }
    }

    public class CartaoJogoDto
    {
        public int JogadorId { get; set; }
        public string JogadorNome { get; set; } = string.Empty;
        public int Minuto { get; set; }
        public string Tipo { get; set; } = string.Empty; // "Amarelo" | "Vermelho"
    }

    public class EstatisticaTimeJogoDto
    {
        public string Nome { get; set; } = string.Empty; // chave da api-football (ex.: "Ball Possession")
        public string? ValorCasa { get; set; }
        public string? ValorVisitante { get; set; }
    }
}

File: AndroidApp/app/src/main/java/com/comentarista/futebol/data/remote/dto/JogoDtos.kt
package com.comentarista.futebol.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class JogoResumoDto(
    val id: Int,
    val data: String? = null,
    val competicaoId: Int,
    val competicaoNome: String? = null,
    val timeCasaId: Int,
    val timeCasaNome: String,
    val timeCasaEscudoUrl: String? = null,
    val timeVisitanteId: Int,
    val timeVisitanteNome: String,
    val timeVisitanteEscudoUrl: String? = null,
    val placarCasa: Int? = null,
    val placarVisitante: Int? = null,
    val status: String? = null,
    val analisadoPorMim: Boolean = false
)

@Serializable
data class JogoDetalheDto(
    val id: Int,
    val data: String? = null,
    val competicaoId: Int,
    val competicaoNome: String? = null,
    val timeCasaId: Int,
    val timeCasaNome: String,
    val timeCasaEscudoUrl: String? = null,
    val timeVisitanteId: Int,
    val timeVisitanteNome: String,
    val timeVisitanteEscudoUrl: String? = null,
    val placarCasa: Int? = null,
    val placarVisitante: Int? = null,
    val status: String? = null,
    val analisadoPorMim: Boolean = false,
    val estadio: String? = null,
    val arbitro: String? = null,
    val penaltisCasa: Int? = null,
    val penaltisVisitante: Int? = null,
    val escalacaoCasa: List<EscalacaoJogadorDto> = emptyList(),
    val escalacaoVisitante: List<EscalacaoJogadorDto> = emptyList(),
    val gols: List<GolJogoDto> = emptyList(),
    val cartoes: List<CartaoJogoDto> = emptyList(),
    val estatisticasTimes: List<EstatisticaTimeJogoDto> = emptyList()
)

@Serializable
data class EscalacaoJogadorDto(
    val jogadorId: Int,
    val nome: String,
    val posicao: String? = null,
    val titular: Boolean = false,
    val fotoUrl: String? = null
)

@Serializable
data class GolJogoDto(
    val jogadorId: Int,
    val jogadorNome: String,
    val minuto: Int,
    val contra: Boolean = false
)

@Serializable
data class CartaoJogoDto(
    val jogadorId: Int,
    val jogadorNome: String,
    val minuto: Int,
    val tipo: String
)

@Serializable
data class EstatisticaTimeJogoDto(
    val nome: String,
    val valorCasa: String? = null,
    val valorVisitante: String? = null
)
